python/instrumentation/openinference-instrumentation-dspy/src/openinference/instrumentation/dspy/__init__.py

```diff
@@ -991,6 +993,48 @@ def _output_value_and_mime_type(response: Any) -> Iterator[Tuple[str, Any]]:
 def _llm_model_name(lm: "LM") -> Iterator[Tuple[str, Any]]:
     if (model_name := getattr(lm, "model_name", None)) is not None:
         yield LLM_MODEL_NAME, model_name
+    elif (model := getattr(lm, "model", None)) is not None:
+        # Extract model name from "provider/model" format or use as-is
+        model_str = str(model)
+        if "/" in model_str:
+            # Extract model from "provider/model" format (e.g., "openai/gpt-4" -> "gpt-4")
+            model_name = model_str.split("/", 1)[1]
```

Comment:
what if there is a trailing slash. Is this actually safe? I know this might be how litellm does things but might be worth double checking